src/fmu/tools/rms/generate_petro_jobs_for_field_update.py

```diff
@@ -0,0 +1,506 @@
+
+    if report_unused:
+        # First report which field parameters from original model
+        # that is not specified to be used
+        print("Report of unused petrophysical variables in generated jobs:")
+        for zone_model in zone_models_list:
+            zone_name = zone_model["ZoneName"]
+            if zone_name not in used_petro_per_zone_per_facies_dict:
+                print(f" No field parameters are used from zone: {zone_name}")
+            else:
+                petro_per_facies_dict = used_petro_per_zone_per_facies_dict[zone_name]
+                facies_models_list = zone_model["Facies Models"]
+                for facies_model in facies_models_list:
+                    facies_name = facies_model["FaciesName"]
+                    petro_model_list = facies_model["Variable Models"]
+                    if facies_name not in petro_per_facies_dict:
+                        print(
+                            " No field parameters are used for facies "
+                            f"{facies_name} for zone {zone_name}"
+                        )
+                    else:
+                        petro_list = petro_per_facies_dict[facies_name]
+                        for petro_model in petro_model_list:
+                            var_name = petro_model["VariableName"]
+                            if var_name not in petro_list:
+                                print(
+                                    f" Field parameter {var_name} is not used "
+                                    f"for facies {facies_name} for zone {zone_name}"
+                                )
+        print("")
+
+    # Check if there are specified field parameters which does not exist
+    # in the original job and report errors if this is the case
+    specified_petro_var_list = get_used_petro_names(used_petro_per_zone_per_facies_dict)
+    err_list = []
+    for specified_petro_var in specified_petro_var_list:
+        found = False
+        for zone_model in zone_models_list:
+            facies_models_list = zone_model["Facies Models"]
+            for facies_model in facies_models_list:
+                petro_model_list = facies_model["Variable Models"]
+                for petro_model in petro_model_list:
+                    if specified_petro_var == petro_model["VariableName"]:
+                        found = True
+                        break
+                if found:
+                    break
+            if found:
+                break
+        if not found:
+            err_list.append(specified_petro_var)
+    if len(err_list) > 0:
+        print("Error in specification of used petrophysical variables.")
+        print("Unknown petrophysical variables:")
+        for name in err_list:
+            print(f"{name}")
+        raise ValueError("Unknown petrophysical variable names are specified.")
+
+
+def create_new_petro_job_per_facies(spec_dict, debug_print=False, report_unused=False):
```

Comment:
This is quite long; possible to refactor? (ChatGPT in VScode can be rather useful to help you)

Reply:
Maybe later